File: pm_core_1/src/infrastructure/store/mapper/issue_type_mapper.go
/*
Author: QuanTuanHuy
Description: Part of Serp Project
*/

package mapper

import (
	"github.com/serp/pm-core/src/core/domain/entity"
	"github.com/serp/pm-core/src/infrastructure/store/model"
)

type IssueTypeMapper struct{}

func NewIssueTypeMapper() *IssueTypeMapper {
	return &IssueTypeMapper{}
}

func (m *IssueTypeMapper) ToEntity(mdl *model.IssueTypeModel) *entity.IssueTypeEntity {
	if mdl == nil {
		return nil
	}

	return &entity.IssueTypeEntity{
		BaseEntity: entity.BaseEntity{
			ID:        mdl.ID,
			CreatedAt: mdl.CreatedAt.UnixMilli(),
			CreatedBy: mdl.CreatedBy,
			UpdatedAt: mdl.UpdatedAt.UnixMilli(),
			UpdatedBy: mdl.UpdatedBy,
		},
		TenantID:       mdl.TenantID,
		TypeKey:        mdl.TypeKey,
		Name:           mdl.Name,
		Description:    mdl.Description,
		IconUrl:        mdl.IconUrl,
		HierarchyLevel: mdl.HierarchyLevel,
		IsSystem:       mdl.IsSystem,
	}
}

func (m *IssueTypeMapper) ToModel(e *entity.IssueTypeEntity) *model.IssueTypeModel {
	if e == nil {
		return nil
	}
	return &model.IssueTypeModel{
		BaseModel: model.BaseModel{
			ID:        e.ID,
			UpdatedBy: e.UpdatedBy,
		},
		TenantID:       e.TenantID,
		TypeKey:        e.TypeKey,
		Name:           e.Name,
		Description:    e.Description,
		IconUrl:        e.IconUrl,
		HierarchyLevel: e.HierarchyLevel,
		IsSystem:       e.IsSystem,
	}
}

func (m *IssueTypeMapper) ToEntities(models []*model.IssueTypeModel) []*entity.IssueTypeEntity {
	if models == nil {
		return nil
	}
	entities := make([]*entity.IssueTypeEntity, 0, len(models))
	for _, mdl := range models {
		if e := m.ToEntity(mdl); e != nil {
			entities = append(entities, e)
		}
	}
	return entities
}

File: pm_core/src/main/java/serp/project/pmcore/infrastructure/store/mapper/IssueTypeMapper.java
/**
 * Author: QuanTuanHuy
 * Description: Part of Serp Project
 */

package serp.project.pmcore.infrastructure.store.mapper;

import org.springframework.stereotype.Component;
import serp.project.pmcore.core.domain.entity.IssueTypeEntity;
import serp.project.pmcore.infrastructure.store.model.IssueTypeModel;

import java.util.Collections;
import java.util.List;
import java.util.stream.Collectors;

@Component
public class IssueTypeMapper extends BaseMapper {

    public IssueTypeEntity toEntity(IssueTypeModel model) {
        if (model == null) { return null; }
        return IssueTypeEntity.builder()
                .id(model.getId())
                .tenantId(model.getTenantId())
                .typeKey(model.getTypeKey())
                .name(model.getName())
                .description(model.getDescription())
                .iconUrl(model.getIconUrl())
                .hierarchyLevel(model.getHierarchyLevel())
                .isSystem(Boolean.TRUE.equals(model.getIsSystem()))
                .createdAt(localDateTimeToLong(model.getCreatedAt()))
                .createdBy(model.getCreatedBy())
                .updatedAt(localDateTimeToLong(model.getUpdatedAt()))
                .updatedBy(model.getUpdatedBy())
                .build();
    }

    public IssueTypeModel toModel(IssueTypeEntity entity) {
        if (entity == null) { return null; }
        return IssueTypeModel.builder()
                .id(entity.getId())
                .tenantId(entity.getTenantId())
                .typeKey(entity.getTypeKey())
                .name(entity.getName())
                .description(entity.getDescription())
                .iconUrl(entity.getIconUrl())
                .hierarchyLevel(entity.getHierarchyLevel())
                .isSystem(entity.isSystem())
                .createdAt(longToLocalDateTime(entity.getCreatedAt()))
                .createdBy(entity.getCreatedBy())
                .updatedAt(longToLocalDateTime(entity.getUpdatedAt()))
                .updatedBy(entity.getUpdatedBy())
                .build();
    }

    public List<IssueTypeEntity> toEntities(List<IssueTypeModel> models) {
        if (models == null) { return Collections.emptyList(); }
        return models.stream().map(this::toEntity).collect(Collectors.toList());
    }
}
